src/CoreShop/Behat/Context/Transform/CartContext.php

```diff
@@ -25,26 +26,46 @@
      */
     private $sharedStorage;
 
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
     /**
      * @var CartContextInterface
      */
     private $cartContext;
 
     /**
      * @param SharedStorageInterface $sharedStorage
+     * @param CartRepositoryInterface $cartRepository
      * @param CartContextInterface $cartContext
      */
-    public function __construct(SharedStorageInterface $sharedStorage, CartContextInterface $cartContext)
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        CartRepositoryInterface $cartRepository,
+        CartContextInterface $cartContext
+    )
     {
         $this->sharedStorage = $sharedStorage;
+        $this->cartRepository = $cartRepository;
         $this->cartContext = $cartContext;
     }
 
     /**
      * @Transform /^my cart/
+     * @Transform /^cart(?:|s)/
      */
     public function cart()
     {
         return $this->cartContext->getCart();
     }
+
+    /**
+     * @Transform /^loaded cart(?:|s)/
+     */
+    public function loadedCart()
```

Comment:
just to be clear here, the `loaded cart` is to test if the value gets persisted right. If we don't do that, we can set any value we want and the test would pass. even string values would be accepted by Pimcore. So, once we force load the object, we get the newly hydrated object and can test against that.